src/test/mir-opt/retag.array_casts.SimplifyCfg-elaborate-drops.after.mir

```diff
@@ -128,7 +128,12 @@ fn array_casts() -> () {
         Retag(_35);                      // scope 5 at $SRC_DIR/core/src/macros/mod.rs:LL:COL
         _18 = &(*_35);                   // scope 5 at $SRC_DIR/core/src/macros/mod.rs:LL:COL
         Retag(_18);                      // scope 5 at $SRC_DIR/core/src/macros/mod.rs:LL:COL
-        _13 = (move _14, move _18);      // scope 5 at $SRC_DIR/core/src/macros/mod.rs:LL:COL
+        Deinit(_13);                     // scope 5 at $SRC_DIR/core/src/macros/mod.rs:LL:COL
+        (_13.0: &usize) = move _14;      // scope 5 at $SRC_DIR/core/src/macros/mod.rs:LL:COL
+        Retag((_13.0: &usize));          // scope 5 at $SRC_DIR/core/src/macros/mod.rs:LL:COL
+        (_13.1: &usize) = move _18;      // scope 5 at $SRC_DIR/core/src/macros/mod.rs:LL:COL
+        Retag((_13.1: &usize));          // scope 5 at $SRC_DIR/core/src/macros/mod.rs:LL:COL
+        Finalize(_13);                   // scope 5 at $SRC_DIR/core/src/macros/mod.rs:LL:COL
```

Comment:
The fact that we now emit the additional retags is interesting. I *think* this is the behavior that we always should have had, but I'm not sure. cc @RalfJung

Reply:
jup, there are now some minor diagnostics changes in miri where we point to more precise locations

Reply:
Miri currently only retags bare references, not references inside aggregrates. This change here "surfaced" more bare references, hence there are more retags.